neo/rawio/spikegadgetsrawio.py

```diff
@@ -79,6 +80,24 @@ def __init__(self, filename="", selected_streams=None):
     def _source_name(self):
         return self.filename
 
+    def _produce_ephys_channel_ids(self, n_total_channels, n_channels_per_chip):
+        """Compute the channel ID labels
+        The ephys channels in the .rec file are stored in the following order:
+        hwChan ID of channel 0 of first chip, hwChan ID of channel 0 of second chip, ..., hwChan ID of channel 0 of Nth chip,
+        hwChan ID of channel 1 of first chip, hwChan ID of channel 1 of second chip, ..., hwChan ID of channel 1 of Nth chip,
+        ...
+        So if there are 32 channels per chip and 128 channels (4 chips), then the channel IDs are:
+        0, 32, 64, 96, 1, 33, 65, 97, ..., 128
+        See also: https://github.com/NeuralEnsemble/python-neo/issues/1215
+        """
+        ephys_channel_ids_list = []
+        for hw_channel in range(n_channels_per_chip):
+            hw_channel_list = [
+                hw_channel + chip * n_channels_per_chip for chip in range(int(n_total_channels / n_channels_per_chip))
```

Comment:
will this int(n_total_channels / n_channels_per_chip) works for any n_total_channels numbers.
Do we need an explicit ceil or  floor ?

Reply:
I think we should be good. Most of the times you'll have increments of 32 channels/chip. So total channels would have to be some multiple of 32. I don't know gadgets well enough, but I guess the problem would be Intan does sell 16 channel headstage so if you were allowed to mix headstage chips then this math wouldn't work. But I think in this case we wait for a user to open an issue so we have mixed headstage data to work with.